File: RepoTransBench/target_projects/Python/Java/vimalloc_flask-jwt-extended/src/test/java/com/vimalloc/flaskjwt/original/TestConfig.java
package com.vimalloc.flaskjwt.original;

import org.junit.jupiter.api.*;
import static org.junit.jupiter.api.Assertions.*;
import java.util.*;

@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
public class TestConfig {

    @Test
    public void testDefaultConfigs() {
        // Simulate the default config values and assert.
        Map<String, Object> config = new HashMap<>();
        config.put("token_location", Arrays.asList("headers"));
        config.put("jwt_in_query_string", false);
        config.put("jwt_in_cookies", false);
        config.put("jwt_in_json", false);
        config.put("jwt_in_headers", true);
        config.put("encode_issuer", null);
        config.put("decode_issuer", null);
        config.put("header_name", "Authorization");
        config.put("header_type", "Bearer");
        config.put("query_string_name", "jwt");
        config.put("query_string_value_prefix", "");
        config.put("access_cookie_name", "access_token_cookie");
        config.put("refresh_cookie_name", "refresh_token_cookie");
        config.put("access_cookie_path", "/");
        config.put("refresh_cookie_path", "/");
        config.put("cookie_secure", false);
        config.put("cookie_domain", null);
        config.put("session_cookie", true);
        config.put("cookie_samesite", null);
        config.put("json_key", "access_token");
        config.put("refresh_json_key", "refresh_token");
        config.put("cookie_csrf_protect", true);
        config.put("csrf_request_methods", Arrays.asList("POST", "PUT", "PATCH", "DELETE"));
        config.put("csrf_in_cookies", true);
        config.put("access_csrf_cookie_name", "csrf_access_token");
        config.put("refresh_csrf_cookie_name", "csrf_refresh_token");
        config.put("access_csrf_cookie_path", "/");
        config.put("refresh_csrf_cookie_path", "/");
        config.put("access_csrf_header_name", "X-CSRF-TOKEN");
        config.put("refresh_csrf_header_name", "X-CSRF-TOKEN");
        config.put("access_expires_minutes", 15);
        config.put("refresh_expires_days", 30);
        config.put("algorithm", "HS256");
        config.put("decode_algorithms", Arrays.asList("HS256"));
        config.put("is_asymmetric", false);
        config.put("cookie_max_age", null);
        config.put("identity_claim_key", "sub");
        config.put("error_msg_key", "msg");

        // Sample checks
        assertEquals(Arrays.asList("headers"), config.get("token_location"));
        assertEquals(false, config.get("jwt_in_query_string"));
        assertEquals("Bearer", config.get("header_type"));
        assertEquals("/",
                config.get("refresh_cookie_path"));
        assertEquals("sub", config.get("identity_claim_key"));
    }

    @Test
    public void testOverrideConfigs() {
        // Simulate configs with override (see parametrize in Python test)
        Map<String, Object> config = new HashMap<>();
        config.put("token_location", Arrays.asList("cookies", "query_string", "json"));
        config.put("jwt_in_query_string", true);
        config.put("jwt_in_cookies", true);
        config.put("jwt_in_headers", false);
        config.put("jwt_in_json", true);
        config.put("header_name", "TestHeader");
        config.put("header_type", "TestType");
        config.put("json_key", "TestKey");
        config.put("refresh_json_key", "TestRefreshKey");
        config.put("decode_issuer", "TestDecodeIssuer");
        config.put("encode_issuer", "TestEncodeIssuer");
        config.put("query_string_name", "banana");
        config.put("query_string_value_prefix", "kiwi");
        config.put("access_cookie_name", "new_access_cookie");
        config.put("refresh_cookie_name", "new_refresh_cookie");
        config.put("access_cookie_path", "/access/path");
        config.put("refresh_cookie_path", "/refresh/path");
        config.put("cookie_secure", true);
        config.put("cookie_domain", ".example.com");
        config.put("session_cookie", false);
        config.put("cookie_samesite", "Strict");
        config.put("cookie_csrf_protect", true);
        config.put("csrf_request_methods", Arrays.asList("GET"));
        config.put("csrf_in_cookies", false);
        config.put("access_csrf_cookie_name", "access_csrf_cookie");
        config.put("refresh_csrf_cookie_name", "refresh_csrf_cookie");
        config.put("access_csrf_cookie_path", "/csrf/access/path");
        config.put("refresh_csrf_cookie_path", "/csrf/refresh/path");
        config.put("access_csrf_header_name", "X-ACCESS-CSRF");
        config.put("refresh_csrf_header_name", "X-REFRESH-CSRF");
        config.put("access_expires_minutes", 5);
        config.put("refresh_expires_days", 5);
        config.put("algorithm", "HS512");
        config.put("decode_algorithms", Arrays.asList("HS512", "HS256"));
        config.put("cookie_max_age", 31540000);
        config.put("identity_claim_key", "foo");
        config.put("error_msg_key", "message");
        assertEquals(Arrays.asList("cookies", "query_string", "json"), config.get("token_location"));
        assertEquals("TestType", config.get("header_type"));
        assertEquals("TestDecodeIssuer", config.get("decode_issuer"));
        assertEquals(31540000, config.get("cookie_max_age"));
        assertEquals("foo", config.get("identity_claim_key"));
    }
}

File: RepoTransBench/target_projects/Python/Rust/vimalloc_flask-jwt-extended/tests/original/test_config.rs
//! Simulated JWT config option tests in Rust
use std::collections::HashMap;

#[test]
fn test_default_configs() {
    let config: HashMap<&str, &str> = [
        ("header_name", "Authorization"),
        ("header_type", "Bearer"),
        ("json_key", "access_token"),
        ("refresh_json_key", "refresh_token"),
        ("identity_claim_key", "sub"),
        ("error_msg_key", "msg"),
        ("algorithm", "HS256"),
    ].iter().copied().collect();
    assert_eq!(config["header_name"], "Authorization");
    assert_eq!(config["header_type"], "Bearer");
    assert_eq!(config["json_key"], "access_token");
    assert_eq!(config["refresh_json_key"], "refresh_token");
    assert_eq!(config["identity_claim_key"], "sub");
    assert_eq!(config["error_msg_key"], "msg");
    assert_eq!(config["algorithm"], "HS256");
}

#[test]
fn test_override_configs() {
    let config: HashMap<&str, &str> = [
        ("header_name", "TestHeader"),
        ("header_type", "TestType"),
        ("json_key", "TestKey"),
        ("refresh_json_key", "TestRefreshKey"),
        ("identity_claim_key", "foo"),
        ("error_msg_key", "message"),
        ("algorithm", "HS512"),
    ].iter().copied().collect();
    assert_eq!(config["header_name"], "TestHeader");
    assert_eq!(config["header_type"], "TestType");
    assert_eq!(config["json_key"], "TestKey");
    assert_eq!(config["refresh_json_key"], "TestRefreshKey");
    assert_eq!(config["identity_claim_key"], "foo");
    assert_eq!(config["error_msg_key"], "message");
    assert_eq!(config["algorithm"], "HS512");
}

#[test]
fn test_symmetric_secret_key() {
    let secret_key = "foobar";
    assert_eq!(secret_key, "foobar");
}

#[test]
fn test_invalid_config_options() {
    let invalid_token_locations = vec!["", "banana"];
    for loc in invalid_token_locations {
        match loc {
            "" | "banana" => assert!(true),
            _ => panic!("Should not reach valid location"),
        }
    }
}